### PR TITLE
Add ability to specify name in file extension client

### PR DESCRIPTION
**Pull request by pmm-sumo**
**Description:** 

This is a proposal how to make File Extension more generic. When used e.g. for WAL purposes this will allow to have separate client for each signal type

### CODE REVIEW

**Comment by pmm-sumo** [comment]
@djaglowski this is a quick draft that adds `name` as another attribute to `GetClient`, giving a generic ability to use separate storage instances for the same type of component. What do you think?

File: extension/storage/README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ A storage extension persists state beyond the collector process. Other component
 
 The `storage.Extension` interface extends `component.Extension` by adding the following method:
 ```
-GetClient(component.Kind, component.Kind, config.ComponentID) (Client, error)
+GetClient(component.Kind, component.Kind, config.ComponentID, string) (Client, error)
```
Comment (djaglowski) [inline extension/storage/README.md:7]: Looks like there was a preexisting typo here.
```suggestion
GetClient(context.Context, component.Kind, config.ComponentID, string) (Client, error)
```

**Review by djaglowski — APPROVED** [review]
Looks good to me.

**Comment by pmm-sumo** [comment]
I have added a simple test-case for single component having two clients and addressed the README.md fix. Build seems to fail on an unrelated load-test at the moment

**Pull request merged**